File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace GridViewGroupSelectionMvc
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView(NorthwindDataProvider.GetProducts())
		End Function

		Public Function GridViewCustomActionPartial(ByVal parameters As String) As ActionResult
			ViewData("parameters") = parameters
			Return PartialView("GridViewPartial", NorthwindDataProvider.GetProducts())
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using Models;
using System.Web.Mvc;

namespace GridUpdateComboMvc
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        [ValidateInput(false)]
        public ActionResult GridViewPartial()
        {
            return PartialView("_GridViewPartial", BatchEditRepository.GridData);
        }

        public ActionResult GridViewCustomAction(string parameters)
        {
            ViewData["data"] = parameters;
            return GridViewPartial();
        }

    }
}